src/main/java/io/quarkus/gizmo2/creator/ops/StringBuilderOps.java

```diff
@@ -0,0 +1,83 @@
+ * <li>Append to it using {@link #append(Expr)}</li>
+ * <li>Create the final string using {@link #objToString()}</li>
+ * </ol>
+ */
+public final class StringBuilderOps extends ObjectOps implements ComparableOps {
+
+    /**
+     * Construct a new instance on an existing {@link StringBuilder}.
+     *
+     * @param bc the block creator (must not be {@code null})
+     * @param obj the receiver string builder (must not be {@code null})
+     */
+    public StringBuilderOps(final BlockCreator bc, final Expr obj) {
+        super(StringBuilder.class, bc, obj);
+    }
+
+    /**
+     * Construct a new instance on a newly created {@link StringBuilder}.
+     * The {@code StringBuilder} instance may be obtained using {@link #getInstance()}.
+     *
+     * @param bc the block creator (must not be {@code null})
+     */
+    public StringBuilderOps(final BlockCreator bc) {
+        super(StringBuilder.class, bc, bc.define("stringBuilder", bc.new_(StringBuilder.class)));
+    }
+
+    /**
+     * Construct a new instance on a newly created {@link StringBuilder}
+     * with given {@code capacity}. The {@code StringBuilder} instance
+     * may be obtained using {@link #getInstance()}.
+     *
+     * @param bc the block creator (must not be {@code null})
+     * @param capacity the capacity of the newly created {@link StringBuilder}
+     */
+    public StringBuilderOps(final BlockCreator bc, final int capacity) {
+        super(StringBuilder.class, bc, bc.new_(StringBuilder.class, Constant.of(capacity)));
+    }
+
+    public StringBuilderOps append(final Expr expr) {
+        switch (expr.type().descriptorString()) {
+            case "Z" -> invokeInstance(StringBuilder.class, "append", boolean.class, expr);
+            case "B", "S", "I" -> invokeInstance(StringBuilder.class, "append", int.class, expr);
+            case "J" -> invokeInstance(StringBuilder.class, "append", long.class, expr);
+            case "F" -> invokeInstance(StringBuilder.class, "append", float.class, expr);
+            case "D" -> invokeInstance(StringBuilder.class, "append", double.class, expr);
+            case "C" -> invokeInstance(StringBuilder.class, "append", char.class, expr);
+            case "[C" -> invokeInstance(StringBuilder.class, "append", char[].class, expr);
+            case "Ljava/lang/String;" -> invokeInstance(StringBuilder.class, "append", String.class, expr);
+            case "Ljava/lang/CharSequence;" -> invokeInstance(StringBuilder.class, "append", CharSequence.class, expr);
+            default -> invokeInstance(StringBuilder.class, "append", Object.class, expr);
+        }
+        return this;
+    }
+
+    public StringBuilderOps append(final char constant) {
+        return append(Constant.of(constant));
+    }
+
+    public StringBuilderOps append(final String constant) {
+        return append(Constant.of(constant));
+    }
+
+    public Expr getInstance() {
```

Comment:
I wonder if we shouldn't just make the `ObjectOps#obj` field public instead... :thinking:

Reply:
> I'm not a huge fan of public fields; at most it should be an `instance()` method on the base.

Me neither, except for `public final` fields.

@Ladicek why do we need this method for `StringBuilderOps` anyway?

Reply:
I don't actually think we need it, because there's a way to create `StringBuilderOps` on an existing `StringBuilder`, so if someone wants to access it from the outside, they can use that. (That possibility is new and wasn't present in Gizmo 1.x.) So I can just remove `instance()` if you agree.

Reply:
Done, removed the method.